### ISSUE TITLE
Unused route table created when creating a subnet

### ISSUE THREAD

**Issue by Kent1** [aws/aws-cdk#3604]
* **I'm submitting a ...**
  - [x] :beetle: bug report
  - [ ] :rocket: feature request
  - [ ] :books: construct library gap
  - [ ] :phone: security issue or vulnerability => Please see [policy](https://github.com/aws/aws-cdk/security/policy)
  - [ ] :question: support request => Please see note at the top of this template.

* **What is the current behavior?**
When i create a Subnet, and change the RouteTableAssociation to an existing route table, an unused route table is created along with the subnet.

* **What is the expected behavior (or behavior of feature suggested)?**
I would like to be able to remove this route table or that cdk is smart enough to not create it

* **Please tell us about your environment:**

  - CDK CLI Version: 1.3.0
  - Module Version: 1.3.0
  - OS: [OSX Mojave]
  - Language: [Java]

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. associated pull-request, stackoverflow, gitter, etc)

```java
Subnet Subnet = new Subnet(this, "Subnet", SubnetProps.builder()
        .withVpcId(vpc.getVpcId())
        .withAvailabilityZone("eu-west-1b")
        .withCidrBlock(String.format("10.%d.43.128/25", environmentId))
        .build());

CfnSubnetRouteTableAssociation routeTableAssociation = (CfnSubnetRouteTableAssociation) subnet.getNode().tryFindChild("RouteTableAssociation");
routeTableAssociation.setRouteTableId(Fn.importValue(String.format("%s-nat-nat000", environment)));

CfnRouteTable cfnRouteTable = (CfnRouteTable) subnet.getNode().tryFindChild("RouteTable");
```

**Comment by rix0rrr** [comment]
I'm not sure this counts as a bug.

Unfortunately, the `Subnet` construct right now is heavily tied to the `Vpc` construct, and supports exactly the features it needs.

This sounds like a feature request for an optional `RouteTable` input parameter to `Subnet`.

**Comment by konstantinj** [comment]
#4308  RouteTable input would be appreciated.
Not only in SubnetProps but also in subnetConfiguration so it can be used directly in the VPC construct.

**Comment by rix0rrr** [comment]
Related to https://github.com/aws/aws-cdk/issues/5927

**Comment by Obirah** [comment]
Does anyone know an elegant way to "downgrade" from `Subnet` to `CfnSubnet` without having to destroy the pre-existing subnets so that I can get rid of the unused route table? The only way I can think of, is to assign the exact same logical IDs to the `CfnSubnets` that the `Subnet`  construct generated but that is very nasty imho.

**Comment by Obirah** [comment]
In TypeScript I was able to get rid of the RouteTable via `subnet.node.tryRemoveChild('RouteTable');`.

**Comment by japerwu** [comment]
I wish this feature is there. `subnet.node.tryRemoveChild('RouteTable')` results in unresolved dependency in my CFN template as the subnet still tries to reference it.

**Comment by maggoun** [comment]
I want this feature, the ability to assign a route table to a created subnet

**Comment by Vulonus** [comment]
i second this. i need a functionality like this. can we reopen the issue?

**Comment by Vulonus** [comment]
nvm this could be fixed when implementing #4308
